src/main/java/com/google/firebase/cloud/StorageClient.java

```diff
@@ -0,0 +1,144 @@
+ */
+public class StorageClient {
+
+  private static final String DOWNLOAD_TOKENS = "firebaseStorageDownloadTokens";
+
+  private final FirebaseApp app;
+  private final Storage storage;
+
+  private StorageClient(FirebaseApp app) {
+    this.app = checkNotNull(app, "FirebaseApp must not be null");
+    this.storage = StorageOptions.newBuilder()
+        .setCredentials(new FirebaseOAuthCredentials(app))
+        .build()
+        .getService();
+  }
+
+  public static StorageClient getInstance() {
+    return getInstance(FirebaseApp.getInstance());
+  }
+
+  public static StorageClient getInstance(FirebaseApp app) {
+    StorageClientService service = ImplFirebaseTrampolines.getService(app, SERVICE_ID,
+        StorageClientService.class);
+    if (service == null) {
+      service = ImplFirebaseTrampolines.addService(app, new StorageClientService(app));
+    }
+    return service.getInstance();
+  }
+
+  /**
+   * Returns the default cloud storage bucket associated with the current app. This is the bucket
+   * configured via {@link com.google.firebase.FirebaseOptions} when initializing the app. If
+   * no bucket was configured via options, this method throws an exception.
+   *
+   * @return a cloud storage Bucket instance, or null if the configured bucket does not exist.
+   * @throws IllegalArgumentException If no bucket is configured via <code>FirebaseOptions</code>,
+   *     or if the bucket does not exist.
+   */
+  public Bucket getBucket() {
+    return getBucket(app.getOptions().getStorageBucket());
+  }
+
+  /**
+   * Returns a cloud storage Bucket instance for the specified bucket name.
+   *
+   * @param name a non-null, non-empty bucket name.
+   * @return a cloud storage Bucket instance, or null if the specified bucket does not exist.
+   * @throws IllegalArgumentException If the bucket name is null, empty, or if the specified
+   *     bucket does not exist.
+   */
+  public Bucket getBucket(String name) {
+    checkArgument(!Strings.isNullOrEmpty(name),
+        "Bucket name not specified. Specify the bucket name via the storageBucket "
+            + "option when initializing the app, or specify the bucket name explicitly when "
+            + "calling the getBucket() method.");
+    Bucket bucket = storage.get(name);
+    checkArgument(bucket != null, "Bucket " + name + " does not exist.");
+    return bucket;
+  }
+
+  /**
+   * Returns a long-lived download URL with a revokable token. This can be used to share a file
+   * with others, but can be revoked by a developer in the Firebase console if desired.
+   *
+   * @param blob The blob to generate a download URL for.
+   * @return a URL string.
+   */
+  public String getDownloadUrl(Blob blob) {
+    checkNotNull(blob, "Blob must not be null");
+    Map<String, String> metadata = blob.getMetadata();
+    if (metadata == null) {
+      metadata = new HashMap<>();
+    }
+    String downloadTokens = metadata.get(DOWNLOAD_TOKENS);
```

Comment:
Cool, and since this key isn't listed in https://firebase.google.com/docs/storage/web/file-metadata#file_metadata_properties I assume normal Firebase Storage clients can't write it.